src/main/java/algorithms/curated170/medium/LongestSubstringwithAtMostKDistinctCharacters.java

```diff
@@ -0,0 +1,40 @@
+package algorithms.curated170.medium;
+  
+
+public class LongestSubstringwithAtMostKDistinctCharacters {
+
+    public static void main(String[] args) {
+    
+        LongestSubstringwithAtMostKDistinctCharacters lo = new LongestSubstringwithAtMostKDistinctCharacters();
+        int output = lo.lengthOfLongestSubstringKDistinct("abbccc", 2);
+        System.out.println(output);
+    }
+
+    public int lengthOfLongestSubstringKDistinct(String s, int k) {
+       if(k >= s.length()) return s.length();
+      
+        char[] counterMap = new char[256];
```

Comment:
I am not sure, but this might also be possible to implement with a boolean array.

Reply:
Şöyle bir şey, eğer bir char'la karşılaşmamışsak onu görüp kalan değişik harf sayısını azaltıyoruz. 
```
public static int solution(String s, int k)
    {
        int res = 0;
        for (int i = 0; i < s.length(); i++) {
            int r = i+1;
            boolean[] hasLetter = new boolean[256];
            hasLetter[s.charAt(i)] = true;
            int dis = k;
            while(r<s.length() && dis>0)
            {
                if(!hasLetter[s.charAt(r)])
                {
                    dis--;
                    hasLetter[s.charAt(r)] = true;
                }
                r++;
            }
            //System.out.println(i + " " + r);
            r = r==s.length() ? r : r-1;
            res = Math.max(res, r-i);
        }
        return res;
    }
```

Reply:
Bu i'yi dinamik olarak taşıyarak optimize edilebilir.

Reply:
It's actually quite inefficient. It can be considered a naive solution.